### PR TITLE
Solving #19: full configuration of DNS slave IPs authorization

### PR DESCRIPTION
**Pull request by ulvida**
As a PoC (proof of concept), in this first commit 31439f14e4e900983e69da5f12b0247b77c3ab8e, we try the first solution proposed in #19, touching the `/run/alternc/refresh_slave` file after setting DNS slave IP data in the database. 

I'll try the second solution to maintain a better idempotency of the role.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Change log of [ansible role AlternC](https://github.com/UdelaRInterior/ansible-AlternC)
 
+## [v3.2.0-rc1](https://github.com/UdelaRInterior/ansible-AlternC/releases/tag/v3.2.0-rc1)
+
+* Solves issue #15 (but `https_force` variable has no effet on AlternC panel, which is always in https)
+* parially solves #12, and gives a workaround running playbook a second time with tag `alternc_custom`
+* solves wssie 19 with a workaround: a flag to set or not DNS allowed slave IPs, to solve idempotency
```
Comment (andrespias) [inline CHANGELOG.md]: "issue"

**Review by andrespias — COMMENTED** [review]
I've been testing the workround:

> ... a flag to set or not DNS allowed slave IPs, to solve idempotency

And although Alternc has its delays I can say that it works. The proposed code meets its objective.

Setting `alternc_slavedns_bind_ips_configure: yes` only once, we get a relative idempotency while we solve the problem of `slaveip.conf` wrong formed. It is important that its use in defaults is clear.

However, the original proposal:

> to access the web interface of AlternC from ansible, ...

should be explored in another issue.

If we can achive that, we can come up with a higher idempotency solution where we can even leave out the `alternc_slavedns_bind_ips_ansible_exclusive` variable that helps to maintain the relative idempotency.

**Comment by ulvida** [comment]
> I've been testing the workround:
> 
> > ... a flag to set or not DNS allowed slave IPs, to solve idempotency
> 
> And although Alternc has its delays I can say that it works. The proposed code meets its objective.

Ok, good! let's try to release a stable buster version of the role. 

> 
> Setting `alternc_slavedns_bind_ips_configure: yes` only once, we get a relative idempotency while we solve the problem of `slaveip.conf` wrong formed. It is important that its use in defaults is clear.

You think [it's ok](https://github.com/UdelaRInterior/ansible-AlternC/blob/75247dd3c3757a15ac0ec1996ad171021b4dcd72/defaults/main/60_alternc-slavdns.yml#L12)?

> 
> However, the original proposal:
> 
> > to access the web interface of AlternC from ansible, ...
> 
> should be explored in another issue.

I let [the branch `web-api`](https://github.com/UdelaRInterior/ansible-AlternC/tree/web_api) with my last WIP code. I'll put the feature request issue. 

> 
> If we can achive that, we can come up with a higher idempotency solution where we can even leave out the `alternc_slavedns_bind_ips_ansible_exclusive` variable that helps to maintain the relative idempotency.

Well, we can. But I evaluate that the code of such idempotency will be quite heavy, not very ansible oriented, using a lot of "register", as well as complex filters and regex to finally do idempotency tasks, that should be embedded in the modules themselves. It's quite out of my skills, but I think for the same goal, we should better invest in debugging [`alternc-api` code](https://github.com/AlternC/AlternC/tree/master/lib/Alternc/Api).

**Comment by andrespias** [comment]
> > Setting `alternc_slavedns_bind_ips_configure: yes` only once, we get a relative idempotency while we solve the problem of `slaveip.conf` wrong formed. It is important that its use in defaults is clear.
> 
> You think [it's ok](https://github.com/UdelaRInterior/ansible-AlternC/blob/75247dd3c3757a15ac0ec1996ad171021b4dcd72/defaults/main/60_alternc-slavdns.yml#L12)?

Yes, I think it's ok.
 
> > However, the original proposal:
> > > to access the web interface of AlternC from ansible, ...
> > should be explored in another issue.
> 
> I let [the branch `web-api`](https://github.com/UdelaRInterior/ansible-AlternC/tree/web_api) with my last WIP code. I'll put the feature request issue.

Ok, now I see. For some reason I understood that in [`web-api`](https://github.com/UdelaRInterior/ansible-AlternC/tree/web_api)  branch you have been exclusively explore the use of  [`alternc-api`](https://github.com/AlternC/AlternC/tree/master/lib/Alternc/Api).

> > If we can achive that, we can come up with a higher idempotency solution where we can even leave out the `alternc_slavedns_bind_ips_ansible_exclusive` variable that helps to maintain the relative idempotency.
> 
> Well, we can. But I evaluate that the code of such idempotency will be quite heavy, not very ansible oriented, using a lot of "register", as well as complex filters and regex to finally do idempotency tasks, that should be embedded in the modules themselves. It's quite out of my skills, but I think for the same goal, we should better invest in debugging [`alternc-api` code](https://github.com/AlternC/AlternC/tree/master/lib/Alternc/Api).

Ok that's a path to explore. I also think we can explore screen scrapping with another languaje like Python. Anyway, this technique is GUI dependent ... we would get into another complex solution as well.

**Comment by ulvida** [comment]
MAybe we can resolve this PR merging issue_19 into buster. 

For the buster role, we should moreover check the appropriate configuration of ProFTP accounts generated by AlternC.

**Pull request merged**